indexer/ai/src/main/java/com/google/cloudsearch/ai/CustomSkill.java

```diff
@@ -0,0 +1,138 @@
+package com.google.cloudsearch.ai;
+
+import com.google.common.collect.Multimap;
+import org.json.simple.JSONArray;
+import org.json.simple.JSONObject;
+import org.json.simple.parser.JSONParser;
+import org.json.simple.parser.ParseException;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.OutputStream;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+public class CustomSkill implements AISkill{
+
+    private String aiSkillName = "";
+    private List<OutputMapping> outputMappings = new ArrayList<OutputMapping>();
+    private JSONObject input = new JSONObject();
+    private String url = "";
+
+    @Override
+    public void setAISkillName(String aiSkillName) {
+        this.aiSkillName = aiSkillName;
+    }
+
+    @Override
+    public String getAISkillName() {
+        return this.aiSkillName;
+    }
+
+    @Override
+    public void setOutputMappings(JSONArray outputMapping) {
+
+        Iterator<JSONObject> mappingIterator = outputMapping.iterator();
+        while(mappingIterator.hasNext()){
+            JSONObject mappingObject = mappingIterator.next();
+            OutputMapping obj = new OutputMapping();
+            obj.setPropertyNames((String) mappingObject.get(Constants.configTargetProperty));
+            obj.setSkillOutputField((String) mappingObject.get(Constants.configOutputField));
+            this.outputMappings.add(obj);
+        }
+    }
+
+    @Override
+    public List<OutputMapping> getOutputMappings() {
+        return this.outputMappings;
+    }
+
+    @Override
+    public void setInputs(JSONObject input) {
+        this.input = input;
+    }
+
+    @Override
+    public JSONObject getInputs() {
+
+        return this.input;
+    }
+
+    @Override
+    public void setFilter(JSONObject filter) {
+        //Filters not required for Custom functions
+    }
+
+    @Override
+    public JSONObject getFilter() {
+
+        return null;
+    }
```

Comment:
these also can be part of abstract class BaseSkill

Reply:
Inputs and Filters are skill specific, so I didn't implement this in base class.